packages/main/src/plugin/extension-loader.ts

```diff
@@ -310,22 +312,29 @@ export class ExtensionLoader {
       // collect all extensions from the pluginDirectory folders
       const analyzedPluginsDirectoryExtensions: AnalyzedExtension[] = (
         await Promise.all(pluginDirectories.map(folder => this.analyzeExtension(folder, true)))
-      ).filter(extension => extension !== undefined) as AnalyzedExtension[];
+      ).filter(extension => !extension.error) as AnalyzedExtension[];
       analyzedExtensions.push(...analyzedPluginsDirectoryExtensions);
     }
 
     // now we have all extensions, we can load them
     await this.loadExtensions(analyzedExtensions);
   }
 
-  async analyzeExtension(extensionPath: string, removable: boolean): Promise<AnalyzedExtension | undefined> {
+  async analyzeExtension(extensionPath: string, removable: boolean): Promise<AnalyzedExtension> {
     // do nothing if there is no package.json file
+    let error = undefined;
     if (!fs.existsSync(path.resolve(extensionPath, 'package.json'))) {
-      console.warn(`Ignoring extension ${extensionPath} without package.json file`);
-      return undefined;
+      error = `Ignoring extension ${extensionPath} without package.json file`;
+      console.warn(error);
+      return { error } as AnalyzedExtension;
```

Comment:
here I think it may lead to client's error as the object they'll get is not compliant to the typescript contract.
So while I see no problem of using that pattern in tests, we should not in main code as it may lead to runtime errors.

Reply:
Understood, and I agree that would be better. I've changed it to return a regular AnalyzedExtension now. I didn't want to change the API contract, so added a couple 'fake' values like id and name (felt like these could cause problems if empty), and tried to give reasonable defaults/undefined elsewhere. I wasn't sure if there's a simpler way to do the subscription/dispose(), please review again.